src/config.rs

```diff
@@ -147,6 +147,23 @@ impl Default for RawUpdatesConfig {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+struct RawDirectoriesConfig {
+    #[serde(default)]
+    pub custom_pages_dir: PathBuf,
+}
+
+impl Default for RawDirectoriesConfig {
+    fn default() -> Self {
+        Self {
+            custom_pages_dir: match get_app_root(AppDataType::UserData, &crate::APP_INFO) {
+                Ok(data_path) => data_path.join("pages"),
+                Err(err) => todo!("What to do on fail here?"),
```

Comment:
I think there are two options here. First, you could move this code into a function `RawDirectoriesConfig::new -> Result<Self, ...>` and then have `default` call that and then call `expect` on the result (you have to panic, because you cannot return anything else). Due to the (I assume) low failure rate of `get_app_root`, this would work fine.

Otherwise, you could make `RawConfig` less active and have the field be an option (which would be more close to my understanding of a _raw_ config than the first approach). This would mean that the default  just has `None` and that some code later does the lookup of the app root (when a full-on config is _needed_ and failure should actually be handled).

I am not too sure whether one of these suggestions conflicts with some idiome applied here, but let me know if any of the two seems right to you too

Reply:
Yeah, I agree. A panic might be fine, since `get_app_root` shouldn't usually fail, but if it doesn't complicate the code too much I think I'd go with an `Option`. If it cannot be determined, we could simply skip the "custom patch" logic.

If that makes the page lookup logic too complicated, use `expect()` / `unwrap()` instead.

Reply:
I was able to change it to an Option without too much hassle.